File: JavaScript/Problem008.js
const fs = require('fs')
const file = fs.readFileSync('./TextFiles/Problem008.txt', 'UTF8')
const data = file.split("\n").join('')

const solution = (array) => {
  let maxProduct = 0
  for (var i = 0; i < array.length; ++i) {
    let product = 1
    for (let j = i; j < i + 13; ++j) {
      product *= array[j]
    }
    if (product > maxProduct) {
      maxProduct = product
    }
  }
  return maxProduct
}

console.log(solution(data))

File: Haskell/Problem008.hs
import Data.List.Split
import Data.List

main = do
  content <- readFile "Haskell/Problem008.txt"
  print (maximum [product $ map ((read . (:[]))) b | b <- map (take 13) . filter ((>12) . length) $ (tails . concat $ splitOn "\n" content)])
